flink-metrics/flink-metrics-jmx/src/test/java/org/apache/flink/metrics/jmx/JMXReporterFactoryTest.java

```diff
@@ -19,57 +19,53 @@
 
 import org.apache.flink.management.jmx.JMXService;
 import org.apache.flink.metrics.util.MetricReporterTestUtils;
-import org.apache.flink.util.TestLogger;
 
-import org.junit.After;
-import org.junit.Assert;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
 
 import java.io.IOException;
 import java.util.Properties;
 
-import static org.hamcrest.CoreMatchers.allOf;
-import static org.hamcrest.number.OrderingComparison.greaterThanOrEqualTo;
-import static org.hamcrest.number.OrderingComparison.lessThanOrEqualTo;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Tests for the {@link JMXReporterFactory}. */
-public class JMXReporterFactoryTest extends TestLogger {
+class JMXReporterFactoryTest {
 
-    @After
-    public void shutdownService() throws IOException {
+    @AfterEach
+    void shutdownService() throws IOException {
         JMXService.stopInstance();
     }
 
     @Test
-    public void testPortRangeArgument() {
+    void testPortRangeArgument() {
         Properties properties = new Properties();
         properties.setProperty(JMXReporterFactory.ARG_PORT, "9000-9010");
 
         JMXReporter metricReporter = new JMXReporterFactory().createMetricReporter(properties);
         try {
 
-            Assert.assertThat(
-                    metricReporter.getPort().get(),
-                    allOf(greaterThanOrEqualTo(9000), lessThanOrEqualTo(9010)));
+            assertThat(metricReporter.getPort().get())
```

Comment:
```suggestion
            assertThat(opt).get(as(InstanceOfAssertFactories.INTEGER))
```
Just for info... I'm not a huge fan of AssertJ syntax for Optional, but this also guards against _.empty()_.  I'm also OK with just throwing the NoSuchElementException on failure, like we currently do (for readability).

Reply:
I pushed a new version that imo is nicer since it retains type-safety.